resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/internal/NamingThreadFactory.java

```diff
@@ -0,0 +1,40 @@
+package io.github.resilience4j.bulkhead.internal;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Creates named threads using given prefix.
+ * Is based on {@link java.util.concurrent.Executors.DefaultThreadFactory}.
+ */
+class NamingThreadFactory implements ThreadFactory {
+
+    private final ThreadGroup group;
+    private final AtomicInteger threadNumber = new AtomicInteger(1);
+    private final String prefix;
+
+    NamingThreadFactory(String prefix) {
+        this.group = getThreadGroup();
+        this.prefix = prefix + "-";
```

Comment:
I think we should change it to `bulkhead-prefix-`.
What do you think? Usually the prefix might be a name of a backend system.

Reply:
Do you mean something like `this.prefix = "bulkhead-$name-"`? Makes sense.

Reply:
Yes